Compilation fails with GCC 9.1.0

Hey,

Since we are compiling with warning as errors, compilation fails with the latest GCC in two places:
```
../marian-dev/src/common/config_validator.cpp: In member function ‘void marian::ConfigValidator::validateOptionsTraining() const’:
../marian-dev/src/common/config_validator.cpp:86:40: error: implicitly-declared ‘marian::filesystem::Path& marian::filesystem::Path::operator=(const marian::filesystem::Path&)’ is deprecated [-Werror=deprecated-copy]
   86 |     modelDir = filesystem::currentPath();
      |                                        ^
In file included from ../marian-dev/src/common/config_validator.cpp:5:
../marian-dev/src/common/filesystem.h:31:7: note: because ‘marian::filesystem::Path’ has user-provided ‘marian::filesystem::Path::Path(const marian::filesystem::Path&)’
   31 |       Path(const Path& p) : path{p.path} {}
      |       ^~~~
```

and
```
../marian-dev/src/common/io.cpp: In function ‘void marian::io::addMetaToItems(const string&, const string&, std::vector<marian::io::Item>&)’:
../marian-dev/src/common/io.cpp:66:44: error: implicitly-declared ‘marian::Shape& marian::Shape::operator=(const marian::Shape&)’ is deprecated [-Werror=deprecated-copy]
   66 |   item.shape = Shape({(int)meta.size() + 1});
      |                                            ^
In file included from ../marian-dev/src/common/io_item.h:3,
                 from ../marian-dev/src/common/io.h:4,
                 from ../marian-dev/src/common/io.cpp:1:
../marian-dev/src/common/shape.h:45:3: note: because ‘marian::Shape’ has user-provided ‘marian::Shape::Shape(const marian::Shape&)’
   45 |   Shape(const Shape& shape) : Shape() {
      |   ^~~~~
../marian-dev/src/common/io.cpp: In function ‘void marian::io::loadItemsFromNpz(const string&, std::vector<marian::io::Item>&)’:
../marian-dev/src/common/io.cpp:94:18: error: implicitly-declared ‘marian::Shape& marian::Shape::operator=(const marian::Shape&)’ is deprecated [-Werror=deprecated-copy]
   94 |     item.shape = shape;
      |                  ^~~~~
In file included from ../marian-dev/src/common/io_item.h:3,
                 from ../marian-dev/src/common/io.h:4,
                 from ../marian-dev/src/common/io.cpp:1:
../marian-dev/src/common/shape.h:45:3: note: because ‘marian::Shape’ has user-provided ‘marian::Shape::Shape(const marian::Shape&)’
   45 |   Shape(const Shape& shape) : Shape() {
      |   ^~~~~
```

Cheers,

Nick

--- a/src/common/filesystem.h
+++ b/src/common/filesystem.h
@@ -29,6 +29,7 @@ namespace filesystem {
     public:
       Path() {}
       Path(const Path& p) : path{p.path} {}
+      Path& operator=(const Path& p) = default;
       Path(const std::string& s) : path{s} {}
       Path(const Pathie::Path& p) : path{p} {}
 

--- a/src/common/shape.h
+++ b/src/common/shape.h
@@ -47,6 +47,8 @@ public:
     std::copy(shape.begin(), shape.end(), begin());
   }
 
+  Shape& operator=(const Shape& p) = default;
+
   inline size_t size() const { return shape_.size(); }
 
   void resize(size_t n) { shape_.resize(n, 1); }

